docs/data/material/components/autocomplete/autocomplete.md

```diff
@@ -83,6 +83,11 @@ The prop is designed to cover the primary use case of a **search input** with su
 
 {{"demo": "FreeSolo.js"}}
 
+> ⚠️ Be careful when using the free solo mode with non-string options, as it may cause type mismatch.  
+> The value created by typing into the textbox is always a string, regardless of the type of the options.
+>
+> If you use the `getOptionLabel` prop in free solo mode, you must make sure it accepts both the option type and a string.
```

Comment:
I think that one could argue that it's an edge that doesn't deserve to be listed in the docs.

```suggestion
```

Reply:
Fair point. I'll move it to API description.